### ISSUE TITLE
Where do I put AfterGameJoined()

### ISSUE THREAD

**Issue by friendlytherapist** [ORelio/Minecraft-Console-Client#612]
Ive narrowed it down to `CheckWall()` but where do I put the `AfterGameJoined()` ? 

```
//MCCScript 1.0
MCC.LoadBot(new WatchWall());

//MCCScript Extensions

class WatchWall : ChatBot
{
	int extrablocks = 0;
	int divWall1 = 0;
	int checks = 0;
	
    //border at back
    //direction facing corner
    //cords of first block

    int cX = -6239;
    int cY = 18;
    int cZ = -21604;

    string direction1 = "east";
    int wall1leng = 379;
    
   // string direction2 = "north";
   // int wall2leng = 76;

	Location blockLoc;

    public WatchWall()
    {
        if (!Settings.TerrainAndMovements)
        {
            LogToConsole("WARNING: Terrain handling is disabled in INI file.");
            UnloadBot();
		}
		else
		{
			//div by 100 to get total checks
			divWall1 = wall1leng/100;
			//get remainder
			extrablocks = wall1leng % 100;
			LogToConsole("Wall length divided up to " + divWall1 + " sets of 100 blocks, plus " + extrablocks + " extra.");
			CheckWall();
		}
 		
 	}
    
    public void CheckWall()
    {	
     	for (checks = 0; checks < divWall1; checks++)    
 		{
 			LogToConsole("Check: " + checks);
 		 	blockLoc = new Location(cX, cY, cZ);
        	LogToConsole("Checking block at: " + blockLoc);   
        	Material blockType = GetWorld().GetBlock(blockLoc).Type;
        	switch (blockType)
        	{
            	case Material.Sand:
					LogToConsole("Check passed at: " + blockLoc);
					nextBlock();
                	break;
            	default:
               	 	LogToConsole("Block at " + blockLoc + " is " + blockType);
					SendText("Wallcheck FAILED at " + blockLoc + " block is " + blockType);
                	break;
        	}
 		}
    }
    
    public void nextBlock()
    {
    	switch (direction1)
    	{
    		case "north":
    			LogToConsole("north");
    			break;
    		case "east":
    			LogToConsole("east");
    			break;
    		case "south":
    			LogToConsole("south");
    			break;
    		case "west":
    			LogToConsole("west");
    			break;
    		default:
    			LogToConsole("ERROR NOT A DIRECTION");
    			break;
     	} 
    }
}
```

Error recieved:

> [MCC] [WatchWall] Wall length divided up to 3 sets of 100 blocks, plus 79 extra. 
> [MCC] [WatchWall] Check: 0 
> [MCC] [WatchWall] Checking block at: X:-6239 Y:18 Z:-21604 
> [MCC] [Script] Script 'scripts/k.cs' failed to run (RuntimeError). 
> [MCC] [Script] System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.InvalidOperationException: ChatBot methods should NOT be called in the constructor as API handler is not initialized yet. Override Initialize() or AfterGameJoined() instead to perform initialization tasks.
>   at MinecraftClient.ChatBot.get_Handler () [0x00023] in <d9cbc5e1dac242a589da951d2d7f2f72>:0 
>   at MinecraftClient.ChatBot.GetWorld () [0x00007] in <d9cbc5e1dac242a589da951d2d7f2f72>:0 
>   at ScriptLoader.Script+WatchWall.CheckWall () [0x00062] in <461df4f14c494822b51b9936fdfd3caf>:0 
>   at ScriptLoader.Script+WatchWall..ctor () [0x000bd] in <461df4f14c494822b51b9936fdfd3caf>:0 
>   at ScriptLoader.Script.__run (MinecraftClient.CSharpAPI __apiHandler, System.String[] args) [0x00007] in <461df4f14c494822b51b9936fdfd3caf>:0 
>   at (wrapper managed-to-native) System.Reflection.MonoMethod.InternalInvoke(System.Reflection.MonoMethod,object,object[],System.Exception&)
>   at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0003b] in <14f1e720d53a4a6a8800252bec102319>:0 
>    --- End of inner exception stack trace ---
>   at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00051] in <14f1e720d53a4a6a8800252bec102319>:0 
>   at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) [0x00000] in <14f1e720d53a4a6a8800252bec102319>:0 
>   at MinecraftClient.CSharpRunner.Run (MinecraftClient.ChatBot apiHandler, System.Threading.ManualResetEvent tickHandler, System.String[] lines, System.String[] args, System.Boolean run) [0x002e4] in <d9cbc5e1dac242a589da951d2d7f2f72>:0  
> >
>

**Comment by TheSnoozer** [comment]
As far as I can remember you would need to put that as a method inside your class. As an example see the watch lamp script (https://github.com/ORelio/Minecraft-Console-Client/blob/master/MinecraftClient/config/sample-script-with-world-access.cs#L38) that overwrites the ``Update`` function. The update function in specific will be called every ~100ms (10fps) (see https://github.com/ORelio/Minecraft-Console-Client/blob/master/MinecraftClient/ChatBot.cs#L101), so you could do a movement every 100ms, until you are finished.

E.g.
```C#
//MCCScript 1.0
MCC.LoadBot(new WatchWall());

//MCCScript Extensions

class WatchWall : ChatBot
{
    int extrablocks = 0;
    int divWall1 = 0;
    int checks = 0;
    
    //border at back
    //direction facing corner
    //cords of first block

    int cX = -6239;
    int cY = 18;
    int cZ = -21604;

    string direction1 = "east";
    int wall1leng = 379;
    
   // string direction2 = "north";
   // int wall2leng = 76;

    Location blockLoc;

    public WatchWall()
    {
        if (!Settings.TerrainAndMovements)
        {
            LogToConsole("WARNING: Terrain handling is disabled in INI file.");
            UnloadBot();
        }
    }

    /// Anything you want to initialize your bot, will be called on load by MinecraftCom
    /// This method is called only once, whereas AfterGameJoined() is called once per server join.
    ///
    /// NOTE: Chat messages cannot be sent at this point in the login process.
    /// If you want to send a message when the bot is loaded, use AfterGameJoined.
    
    public override void Initialize()
    {
        LogToConsole("Initialize was called.");
    }

    /// Called after the server has been joined successfully and chat messages are able to be sent.
    /// This method is called again after reconnecting to the server, whereas Initialize() is called only once.
    ///
    /// NOTE: This is not always right after joining the server - if the bot was loaded after logging
    /// in this is still called.
    public override void AfterGameJoined()
    {
        LogToConsole("AfterGameJoined was called.");
    }

    /// Will be called every ~100ms (10fps) if loaded in MinecraftCom
    public override void Update()
    {
        /// TODO implement logic that moves the bot properly
        //div by 100 to get total checks
        divWall1 = wall1leng/100;
        //get remainder
        extrablocks = wall1leng % 100;
        LogToConsole("Wall length divided up to " + divWall1 + " sets of 100 blocks, plus " + extrablocks + " extra.");
        CheckWall();
    }

    private void CheckWall()
    {   
        for (checks = 0; checks < divWall1; checks++)    
        {
            LogToConsole("Check: " + checks);
            blockLoc = new Location(cX, cY, cZ);
            LogToConsole("Checking block at: " + blockLoc);   
            Material blockType = GetWorld().GetBlock(blockLoc).Type;
            switch (blockType)
            {
                case Material.Sand:
                    LogToConsole("Check passed at: " + blockLoc);
                    nextBlock();
                    break;
                default:
                    LogToConsole("Block at " + blockLoc + " is " + blockType);
                    SendText("Wallcheck FAILED at " + blockLoc + " block is " + blockType);
                    break;
            }
        }
    }

    private void nextBlock()
    {
        switch (direction1)
        {
            case "north":
                LogToConsole("north");
                break;
            case "east":
                LogToConsole("east");
                break;
            case "south":
                LogToConsole("south");
                break;
            case "west":
                LogToConsole("west");
                break;
            default:
                LogToConsole("ERROR NOT A DIRECTION");
                break;
        } 
    }
}



```

**Comment by friendlytherapist** [comment]
So the fastest a script can run / update is 100ms?? Thanks

**Comment by friendlytherapist** [comment]
thanks I think I understand

**Comment by TheSnoozer** [comment]
Cool :-) i think I can then go ahead and close this as 'answered'. If you have any other questions or got stuck somewhere feel free to open a new issue.
